lib/mauth/faraday.rb

```diff
@@ -36,9 +36,16 @@ def initialize(request_env)
       end
 
       def attributes_for_signing
-        request_url = @request_env[:url].path
-        request_url = '/' if request_url.empty?
-        @attributes_for_signing ||= { verb: @request_env[:method].to_s.upcase, request_url: request_url, body: @request_env[:body] }
+        @attributes_for_signing ||= begin
+          request_url = @request_env[:url].path
+          request_url = '/' if request_url.empty?
+          {
+            verb: @request_env[:method].to_s.upcase,
+            request_url: request_url,
+            body: @request_env[:body],
+            query_string: @request_env[:url].query
```

Comment:
So we add the query string always ? How would this interact with a receiver with and old mauth client?

Reply:
Not sure I understand your question. For v1 requests `query_string` will be ignored because it's not in the `SIGNATURE_COMPONENTS` constant in `MAuth::Request` class.

Reply:
I think I understand now.
 There is this method called `attributes_for_signing` here but in fact is more like `potential_attributes_we_may_use_for_signing`. Each version will somehow filter the ones not needed